morphia/src/main/java/org/mongodb/morphia/mapping/MappedClass.java

```diff
@@ -176,6 +176,13 @@ public void update() {
      * Discovers interesting (that we care about) things about the class.
      */
     protected void discover() {
+        try {
```

Comment:
Yes, it is loaded but not initialized. Class.forName() forces the initialization and therefore runs the static block in "EntityWithInterestingAnnotation" class (see the tests). I spotted the solution from here: http://stackoverflow.com/questions/3499214/java-static-class-initialization.